### ISSUE TITLE
Compilation Errors for `tcod-rs`-based Project on `x86_64-pc-windows-gnu`

### ISSUE THREAD

**Issue by indiv0** [tomassedovic/tcod-rs#222]
I'm currently trying to get a project of mine (which uses `tcod-rs`) to compile on Appveyor with `x64_64-pc-windows-gnu`.

It works with MSVC.

It compiles fine on GNU as well, but when trying to run doc tests it gives the following error:

```
    error: linking with `gcc` failed: exit code: 1
    note: "gcc" "-Wl,--enable-long-section-names" "-fno-use-linker-plugin"
    "-Wl,--nxcompat" "-nostdlib" "-m64" "C:\\Program Files
    (x86)\\Rust\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\crt2.o" "C:\\Program
    Files (x86)\\Rust\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsbegin.o" "-L"
    "C:\\Program Files (x86)\\Rust\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib"
    "C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\rustdoctest.IFn9rfSyYfPq\\rust_out.0.o"
    "-o"
    "C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\rustdoctest.IFn9rfSyYfPq\\rust_out.exe"
    "-Wl,--gc-sections" "-nodefaultlibs" "-L"
    "C:\\projects\\tcod-window\\target\\debug\\deps" "-L"
    "C:\\projects\\tcod-window\\target\\debug" "-L"
    "C:\\projects\\tcod-window\\target\\debug\\build\\tcod-sys-621017da41165f97\\out"
    "-L" "C:\\Program Files (x86)\\Rust\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib"
    "-Wl,-Bstatic" "-Wl,-Bdynamic"
    "C:\\projects\\tcod-window\\target\\debug\\libtcod_window.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libtcod-a271f719c3238129.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\liblazy_static-5e6d9f365bf63baa.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libtcod_sys-67dc38e7808c55ef.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\liblibc-5eb0f3d71076a4bc.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libpiston-5240d76c1091e933.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libevent_loop-57a81c78ae414265.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libtime-71756e48b8b5b73b.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libkernel32-757937e2863677bb.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libwindow-260fa35c785d201b.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libshader_version-9378f7f19a190113.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libinput-0e7fe52cd87fd87a.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\librustc_serialize-3bc953984ed46e7f.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libviewport-5fbc07238cd4e800.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libbitflags-e87d150db0333415.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libfloat-455aab8f72d19c0a.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libbitflags-d88a99e5825ec8bf.rlib"
    "C:\\projects\\tcod-window\\target\\debug\\deps\\libwinapi-42bc588a308ea664.rlib"
    "-L" "C:\\Program Files (x86)\\Rust\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib"
    "-l" "std-cb705824" "-l" "tcod" "-l" "tcod" "-l" "SDL.dll" "-l" "opengl32" "-l"
    "tcod" "-l" "kernel32" "-l" "gcc_eh" "-l" "ws2_32" "-l" "userenv" "-l" "shell32"
    "-l" "advapi32" "-l" "compiler-rt" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt"
    "-luser32" "-lkernel32" "C:\\Program Files
    (x86)\\Rust\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
    note: ld: cannot find -lSDL.dll

     error: aborting due to previous error
```

I believe this is due to [this](https://github.com/tomassedovic/tcod-rs/blob/9a3b435519270df554cc89ee3f3bdf4f5144f8ec/tcod_sys/build.rs#L203) line.

Shouldn't it say:

```rust
config.flag("-lSDL");
```

instead of:

```rust
config.flag("-lSDL.dll");
```

If I copy `SDL.dll` into my project root directory as `SDL.dll.dll` it compiles fine.

**Comment by indiv0** [comment]
Actually the above is incorrect. It doesn't work if I copy `SDL.dll.dll` into the project root directory, but it does work if I copy it to `C:\Program Files (x86)\Rust\lib\rustlib\x86_64-pc-windows-gnu\lib\SDL.dll.dll`.

**Comment by tomassedovic** [comment]
That's quite strange. Looks like it's not passing the correct `-L` parameter. I don't see it the output you provided.

Unfortunately, I'm travelling until the end of next week and there won't be a lot of opportunities to have a look at this :-(. I'll have a proper when I get back, sorry!

**Comment by indiv0** [comment]
Alright. For your reference, [here](https://ci.appveyor.com/project/indiv0/tcod-window/build/1.0.6) is an example failed build.

I've got it working currently with [this](https://github.com/indiv0/tcod_window/blob/master/appveyor.yml) `appveyor.yml`. Which hackily fixes the build with [this](https://github.com/indiv0/tcod_window/blob/master/appveyor.yml#L42) line.

Since it's working, there's no rush on this issue (take your time), but it definitely seems like unintended behaviour.

**Comment by tomassedovic** [comment]
Hey @indiv0. I'm really sorry this took so long, but I finally got to it and I think I have a fix.

I've cloned your repo and replicated the issue here:

https://ci.appveyor.com/project/tomassedovic/tcod-window/build/1.0.4#L337

And then tested the fix above:

https://ci.appveyor.com/project/tomassedovic/tcod-window/build/1.0.5

So I'm reasonably sure this commit works. Would appreciate if you could confirm it on your side, though. If it's fine, I'll merge it and publish a new release.

**Comment by indiv0** [comment]
No problem.

I've confirmed it works on my end.
Thanks for taking the time!

**Comment by tomassedovic** [comment]
My pleasure :-). I've released the version 0.10.0 which has the fix. Closing this issue.
